Hello @VindeeR 

Thanks for the feedback. I didn't test that, to be honest. We're still missing a config validator #871. I've added a simple check that doesn't allow that

```
[ERROR] Arguments provided: Namespace(version=False, logfile='DEBUG', logconsole='DEBUG', command='run', expid='t007', notransitive=False, update_version=False, start_time=None, start_after=None, run_only_members=None, profile=None, trace=False)
[ERROR] This is the experiment: <t007> which had an issue with the command: <run> and it is currently using the Autosubmit Version: 4.1.16.
 [CRITICAL] Traceback (most recent call last):
  File "/home/dbeltran/Autosubmit/autosubmit/scripts/autosubmit.py", line 103, in main
    return_value = Autosubmit.run_command(args)
  File "/home/dbeltran/Autosubmit/autosubmit/autosubmit.py", line 888, in run_command
    return Autosubmit.run_experiment(args.expid, args.start_time, args.start_after, args.run_only_members,
  File "/home/dbeltran/Autosubmit/autosubmit/autosubmit.py", line 2154, in run_experiment
    job_list, submitter, exp_history, host, as_conf, platforms_to_test, packages_persistence, _ = Autosubmit.prepare_run(
  File "/home/dbeltran/Autosubmit/autosubmit/autosubmit.py", line 1931, in prepare_run
    as_conf.check_conf_files(running_time=True, force_load=True)
  File "/home/dbeltran/Autosubmit/autosubmit/config/configcommon.py", line 1180, in check_conf_files
    self.validate_config(running_time)
  File "/home/dbeltran/Autosubmit/autosubmit/config/configcommon.py", line 1259, in validate_config
    raise AutosubmitCritical(error_msg, 7014)
autosubmit.log.log.AutosubmitCritical: Chunk_ini value (21) cannot be greater than num_chunks value (20)
Error code: 7014

 [CRITICAL] Chunk_ini value (21) cannot be greater than num_chunks value (20) [eCode=7014]
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -16,6 +16,7 @@
 - Fixed exception trace in AutosubmitCritical and AutosubmitError not being displayed to the user #2109
 - Fix `Jobs.running` to case-insensitive #2916
 - Fixed the wallclock validation error message to display the platform's wallclock time instead of the job time twice #2492
+- Fixes chunkini setting not working correctly #2712
 - Fixed recovery and setstatus for experiments with more than > 32k jobs #2894
 - Removed invalid message about the template script showing the error with a difference of ~5 lines #2962
 - Fix queuing time for the first vertical job

--- a/autosubmit/config/configcommon.py
+++ b/autosubmit/config/configcommon.py
@@ -174,7 +174,6 @@ class AutosubmitConfig(object):
         )
         return str(dir_templates)
 
-
     def get_section(self, section: list[str], d_value: Union[str, Any] = "", must_exists=False) -> str:
         """Gets any section.
 
@@ -222,7 +221,6 @@ class AutosubmitConfig(object):
         """
         return self.jobs_data.get(section, {}).get('WCHUNKINC', "")
 
-
     def get_current_user(self, section: str) -> str:
         """Returns the user to be changed from platform config file.
         This function is used by the autosubmit API.
@@ -241,7 +239,6 @@ class AutosubmitConfig(object):
         """
         return self.get_section([section, 'HOST'], "")
 
-
     def get_current_project(self, section: str) -> str:
         """Returns the project to be changed from platform config file.
         This function is used by the autosubmit API.
@@ -250,7 +247,6 @@ class AutosubmitConfig(object):
         :rtype: str
         """
         return self.get_section([section, 'PROJECT'], "")
-
 
     def set_new_user(self, section: str, new_user: str) -> None:
         """Sets new user for given platform.
@@ -283,7 +279,6 @@ class AutosubmitConfig(object):
         open(self._platforms_parser_file, 'w').write(content)
         open(self._platforms_parser_file, 'a').write(content_to_mod)
 
-
     def set_new_host(self, section: str, new_host: str) -> None:
         """Sets new host for given platform
         This function might be used for autosubmit API after complete migration of `AutosubmitConfigParser`.
@@ -314,7 +309,6 @@ class AutosubmitConfig(object):
         open(self._platforms_parser_file, 'w').write(content)
         open(self._platforms_parser_file, 'a').write(content_to_mod)
 
-
     def set_new_project(self, section: str, new_project: str) -> None:
         """Sets new project for given platform
         This function is used by the autosubmit API.
@@ -344,7 +338,6 @@ class AutosubmitConfig(object):
                 "PROJECT_TO:.*", content_to_mod).group(0)[1:], "PROJECT_TO: " + old_project)
         open(self._platforms_parser_file, 'w').write(content)
         open(self._platforms_parser_file, 'a').write(content_to_mod)
-
 
     def show_messages(self) -> bool:
 
@@ -452,10 +445,10 @@ class AutosubmitConfig(object):
 
     @staticmethod
     def _normalize_jobs_in_wrapper(
-        wrapper: str,
-        wrapper_data: dict[str, Any],
-        job_sections: Iterable[str],
-        raise_exception: bool,
+            wrapper: str,
+            wrapper_data: dict[str, Any],
+            job_sections: Iterable[str],
+            raise_exception: bool,
     ) -> None:
         """Normalize the JOBS_IN_WRAPPER field for a wrapper.
 
@@ -1258,6 +1251,13 @@ class AutosubmitConfig(object):
         err_msg = self.validate_wallclock()
         return err_msg
 
+    def _validate_experiment_conf(self) -> str:
+        """Validate the experiment configuration."""
+        chunks = self.experiment_data.get("CONFIG", {}).get("CHUNKS", None)
+        if chunks and chunks.isdigit() and self.get_chunk_ini() > self.get_num_chunks():
+            return f"Chunk_ini value ({self.get_chunk_ini()}) cannot be greater than num_chunks value ({self.get_num_chunks()})"
+        return ""
+
     def validate_config(self, running_time: bool) -> bool:
         """
         Check if the configuration is valid.
@@ -1266,7 +1266,8 @@ class AutosubmitConfig(object):
         :type running_time: bool
         :raises AutosubmitCritical: If any validation error occurs during runtime.
         """
-        error_msg = self.validate_jobs_conf()
+        error_msg = [self._validate_experiment_conf(), self.validate_jobs_conf()]
+        error_msg = "\n".join([msg for msg in error_msg if msg])
         if not error_msg:
             Log.result('Partial configuration validated correctly')
             return True
@@ -2869,7 +2870,7 @@ class AutosubmitConfig(object):
             if isinstance(wrapper, dict) and section in wrapper.get("JOBS_IN_WRAPPER", []):
                 return wrapper
         return {}
-    
+
     def get_cpmip_thresholds(self, job_section: str) -> dict:
         """Returns the CPMIP thresholds for a given job section.
 

--- a/autosubmit/job/job_list.py
+++ b/autosubmit/job/job_list.py
@@ -1112,7 +1112,7 @@ class JobList(object):
                         if not skip:
                             # get max value in distances_of_current_section.values
                             if job.running == "chunk":
-                                if parent.chunk <= (len(chunk_list) - max_distance):
+                                if parent.chunk <= (chunk_list[-1] - max_distance):
                                     skip = False
                         if not skip:
                             problematic_dependencies.add(parent.name)
@@ -1123,7 +1123,7 @@ class JobList(object):
                         problematic_dependencies.add(parent.name)
                         graph.add_edge(parent.name, job.name)
                     if parent.running == "chunk":
-                        if parent.chunk > (len(chunk_list) - max_distance):
+                        if parent.chunk > (chunk_list[-1] - max_distance):
                             graph.add_edge(parent.name, job.name)
         JobList.handle_frequency_interval_dependencies(chunk, chunk_list, date, date_list, dic_jobs, job,
                                                        member,
@@ -1514,7 +1514,7 @@ class JobList(object):
         skip = False
         if dependency.sign == '-':
             if chunk is not None and len(str(chunk)) > 0 and dependency.running == 'chunk':
-                chunk_index = chunk - 1
+                chunk_index = chunk_list.index(chunk)
                 if chunk_index >= dependency.distance:
                     chunk = chunk_list[chunk_index - dependency.distance]
                 else:

--- a/test/integration/commands/run/test_run_local_scenarios.py
+++ b/test/integration/commands/run/test_run_local_scenarios.py
@@ -435,6 +435,38 @@ def test_run_debug(
         assert exc_info.value.code == 7014
 
 
+def test_run_with_chunk_ini_greater_than_one(
+        autosubmit_exp,
+        general_data,
+        prepare_scratch,
+):
+    yaml = YAML(typ='rt')
+    jobs_data = dedent("""\
+        EXPERIMENT:
+            DATELIST: "200001[01-03]"
+            MEMBERS: "fc[00-02]"
+            NUMCHUNKS: '3'
+            CHUNKINI: '2'
+        JOBS:
+            job:
+                SCRIPT: |
+                    echo "Hello World with id=Success"
+                DEPENDENCIES:
+                    SIM-1:
+                SPLITS: 3
+                PLATFORM: LOCAL
+                RUNNING: chunk
+                wallclock: 00:01
+    """)
+    as_exp = autosubmit_exp(experiment_data=general_data | yaml.load(jobs_data), include_jobs=False, create=True)
+    prepare_scratch(expid=as_exp.expid)
+    as_exp.as_conf.set_last_as_command('run')
+
+    exit_code = as_exp.autosubmit.run_experiment(expid=as_exp.expid)
+
+    assert exit_code == 0
+
+
 @pytest.mark.parametrize(
     "jobs_data, expected_db_entries, final_status, get_call_option",
     [

--- a/test/integration/test_utils/misc.py
+++ b/test/integration/test_utils/misc.py
@@ -17,12 +17,11 @@
 
 """Miscellaneous utilities for integration tests."""
 
-from contextlib import suppress
 from pathlib import Path
 from time import sleep, time
 from typing import TYPE_CHECKING
 
-from portalocker import Lock, AlreadyLocked
+from portalocker import Lock, AlreadyLocked, LockException
 
 if TYPE_CHECKING:
     pass
@@ -71,18 +70,17 @@ def wait_locker(file_lock: Path, expect_locked: bool, timeout: int, interval=0.0
         if expect_locked:
             # Check if the file is locked by attempting to acquire it non-blocking
             try:
-                with Lock(str(file_lock), timeout=0):
+                with Lock(str(file_lock), timeout=0, fail_when_locked=True):
                     # Lock acquired, so it's NOT locked — keep waiting
                     pass
-            except AlreadyLocked:
-                return  # Lock is held
+            except (AlreadyLocked, LockException):
+                return  # Lock is held by another process
         else:
             # Wait for the lock to be released
-            with suppress(AlreadyLocked):
-                try:
-                    with Lock(str(file_lock), timeout=0.01):
-                        return  # Lock released
-                except AlreadyLocked:
-                    pass
+            try:
+                with Lock(str(file_lock), timeout=0, fail_when_locked=True):
+                    return  # Lock released
+            except (AlreadyLocked, LockException):
+                pass
 
         sleep(interval)

--- a/test/unit/config/test_validation.py
+++ b/test/unit/config/test_validation.py
@@ -93,6 +93,38 @@ from autosubmit.log.log import AutosubmitCritical
         False,
         id="Empty wallclock"
     ),
+    pytest.param(
+        {
+            "CONFIG": {"CHUNKS": "5"},
+            "EXPERIMENT": {"NUMCHUNKS": 5, "CHUNKINI": "3"},
+        },
+        False,
+        id="chunk_ini within bounds"
+    ),
+    pytest.param(
+        {
+            "CONFIG": {"CHUNKS": "5"},
+            "EXPERIMENT": {"NUMCHUNKS": 5, "CHUNKINI": "6"},
+        },
+        True,
+        id="chunk_ini exceeds num_chunks"
+    ),
+    pytest.param(
+        {
+            "CONFIG": {"CHUNKS": "not-a-digit"},
+            "EXPERIMENT": {"NUMCHUNKS": 5, "CHUNKINI": "10"},
+        },
+        False,
+        id="chunks not a digit skips chunk_ini validation"
+    ),
+    pytest.param(
+        {
+            "CONFIG": {"CHUNKS": "3"},
+            "EXPERIMENT": {"NUMCHUNKS": 3, "CHUNKINI": ""},
+        },
+        False,
+        id="empty chunk_ini defaults to 1 and is valid"
+    ),
 ])
 def test_validate_conf(autosubmit_config, data: dict, must_fail):
     as_conf = autosubmit_config(expid='t000', experiment_data=data)

